### DISCUSSION TITLE
How to use string number as a input name

### DISCUSSION THREAD

**Discussion by wjureczka** [react-hook-form/react-hook-form#2852] [Q&A] [answered]
Hey,
I am trying to achieve form values like this:
{
parentFilters: [],
filter: {
'50': [10, 20, 30],
'60': []
}
}

The '50' and '60' are parent ids for these in array. Code would look like this: https://codesandbox.io/embed/react-hook-form-defaultinitial-form-value-forked-wdix4?file=/src/index.js

How to set object key as a string '50'? Now it is creating an array. Even if I set this by array injection filter['50'] it is still creating an array instead of object

Now values look like this:
{"parentFilters":[],"filters":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,["10","20"],null,null,null,null,null,null,null,null,null,false]}
but I would like to have {"parentFilters":[],"filters": { '50': ["10","20"] }}. Is it an option? Name '50' is only respected as array index I think.

**Comment by bluebill1049** [comment]
https://codesandbox.io/s/react-hook-form-parse-and-format-textarea-furtc
--

**Comment by wjureczka** [comment]
I mean string number as a '50', '100' etc. not just 'number'

**Comment by bluebill1049** [comment, marked as answer]
<img width="961" alt="Screen Shot 2020-09-09 at 4 42 51 pm" src="https://user-images.githubusercontent.com/10513364/92563813-82791180-f2bb-11ea-9989-659ef59a6be5.png">

**Reply by licatajustin** [reply]
@bluebill1049 I'd love to help support this feature in react-hook-form -- iirc, input names that start with numbers adhere to the html spec -- is this something you're looking at for v7 or something the community can help contribute to?

**Reply by bluebill1049** [reply]
Thanks, @licatajustin, need to be careful with `number`, because we are using it for array index.
